### ISSUE TITLE
[Question / Feature Request] Is there a way to get the best performant mode programmatically?

### ISSUE THREAD

**Issue by dJani97** [ajinasokan/flutter_displaymode#8]
I'm considering to include this library in my future Flutter projects to improve UX. Although I have one question.

In a production app, we obviously don't want to ask the user to select a display mode. Is there a best practice to select the most optimal mode automatically? For example I came up with the following method to select the fastest refresh rate without changing resolution:

```dart
Future<DisplayMode> getOptimalDisplayMode() async {
  final List<DisplayMode> supported = await FlutterDisplayMode.supported;
  final DisplayMode current = await FlutterDisplayMode.current;

  final List<DisplayMode> sameResolution = supported
      .where((DisplayMode m) =>
          m.width == current.width && m.height == current.height)
      .toList();

  sameResolution.sort((DisplayMode a, DisplayMode b) =>
      b.refreshRate.compareTo(a.refreshRate));

  final DisplayMode mostOptimalMode = sameResolution.isNotEmpty
      ? sameResolution.first
      : current;

  print('All modes with the current resolution: $sameResolution');
  print('Most optimal mode: $mostOptimalMode');

  return mostOptimalMode;
}
```

Tested and works well on three devices:
- OnePlus 8T
- Huawei P30
- LG V30+

Other than potentially throwing a `PlatformException`, does this method have any possible pitfalls? Can it cause a crash on some devices?

If this method (or something similar) can reliably select the optimal mode on most devices, can you include it in your package?

**Comment by ajinasokan** [comment]
Unfortunately this cannot be done in a deterministically. Android API docs doesn't specify the mode values will be consistent across vendors. So far I have seen multiple irregularities.

1. `FlutterDisplayMode.current` can fail if the vendor sets app in an automatic switching mode. For eg: in Youtube normal screens has higher refresh rate, when you start playing video it switches to 60.
2. Sometimes it fails just because it is not initialised. To figure out the current mode in such situations there is `FlutterDisplayMode.setDefaultMode`.

Also what is "Optimal" is upto the app and content shown in the screen. It can be optimal for battery or user experience. That being said your implementation looks good to me. Just need to handle the exception case when one of the above situations happen. 

I will take a look at it once more to see if I can put together your snippet with these edge cases 👍

**Comment by dJani97** [comment]
@ajinasokan thank you for your response!

You are right, I was using the words "optimal" and "fastest" interchangeably, which isn't correct. By optimal, I always meant fastest.

I created a helper class for finding the fastest display mode in my apps, and sent it out to a couple of colleagues for testing in a demo app. So far it seems to be stable, not crashing on any device, not even on an iPhone.

The class, and it's usage in main.dart: [flutter_displaymode helper class](https://gist.github.com/dJani97/79ce55d1e80fef96e37a8c21ee85e94c)

Do you think that this solution can be used in production, or it needs some more changes to ensure stability?

**Comment by ajinasokan** [comment]
This seems alright. I think I can add this logic to lib as `setHighRefreshRate` and `setLowRefreshRate` functions.

**Comment by dJani97** [comment]
Thank you, I think it would be really useful for anyone out there looking for a higher-level API.

**Comment by ajinasokan** [comment]
This is now available in `0.3.1-nullsafety.0`
